### PR TITLE
[RFR] - Bundle controller rights

### PR DESCRIPTION
**Pull request by mickaelandrieu**
This pull request solves #363.

* Add bundles to mock application
* Avoid configuration of bundles in mock application
* Tests

This pull request depends on https://github.com/backbee/BackBee/pull/366

### CODE REVIEW

File: Rest/Tests/Controller/BundleControllerTest.php [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with BackBee. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @author Charles Rouillon <charles.rouillon@lp-digital.fr>
+ */
+
+namespace BackBee\Rest\Tests\Controller;
+
+use BackBee\Rest\Controller\PageController;
```
Comment (eric-chau) [inline Rest/Tests/Controller/BundleControllerTest.php]: Unused ``use`` statement :p

File: Rest/Tests/Controller/BundleControllerTest.php [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+use BackBee\Rest\Patcher\OperationBuilder;
+use BackBee\Rest\Test\RestTestCase;
+use BackBee\Security\Group;
+use BackBee\Security\User;
+use BackBee\Site\Site;
+
+/**
+ * Test for PageController class.
```
Comment (eric-chau) [inline Rest/Tests/Controller/BundleControllerTest.php]: ``BundleController``, not ``PageController`` ;)

File: Rest/Tests/Controller/BundleControllerTest.php [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+    {
+        $response = $this->sendRequest(self::requestGet('/rest/1/bundle/demo'));
+        $this->assertTrue($response->isOk());
+        $bundleConfig = json_decode($response->getContent(), true);
+        $this->assertInternalType('array', $bundleConfig);
+        $this->assertCount(10, $bundleConfig);
+        $this->assertArrayHasKey('id', $bundleConfig);
+        $this->assertArrayHasKey('name', $bundleConfig);
+        $this->assertArrayHasKey('description', $bundleConfig);
+        $this->assertArrayHasKey('enable', $bundleConfig);
```
Comment (eric-chau) [inline Rest/Tests/Controller/BundleControllerTest.php]: We should also guarantee that array's keys ``config_per_site``, ``category`` and ``thumbnail`` are also returned.

File: Rest/Tests/Controller/BundleControllerTest.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    }
+
+    public function testPatchAction()
+    {
+        $response = $this->sendRequest(self::requestPatch('/rest/1/bundle/demo', (new OperationBuilder())
+            ->replace('name', 'New DemoBundle name')
+            ->replace('description', 'This is a new description')
+            ->getOperations()
+        ));
+        $this->assertTrue($response->isEmpty());
```
Comment (eric-chau) [inline Rest/Tests/Controller/BundleControllerTest.php]: To be more precise, you should perform ``GET /rest/1/bundle/demo`` to test if ``patchAction`` really works.

**Comment by eric-chau** [comment]
:+1:, good contribution, thanks @mickaelandrieu. If you can test more deeper ``BundleController::patchAction`` it would be nice but I know it is not easy due to ``Configurator``.

**Comment by mickaelandrieu** [comment]
@eric-chau nice catch for patchAction (which was buggy).

Now fixed, make a review on this last point please :)

File: Rest/Tests/Controller/BundleControllerTest.php [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+            ->replace('enable', false)
+            ->replace('category', 'New category')
+            ->getOperations()
+        ));
+        $this->assertTrue($response->isEmpty());
+
+        /* let's check patch actions have been applied */
+        $demoBundle = $this->getBBApp()->getBundle('demo');
+        $this->assertFalse($demoBundle->isEnabled());
+        $this->assertEquals('New category', $demoBundle->getProperty('category')[0]);
```
Comment (eric-chau) [inline Rest/Tests/Controller/BundleControllerTest.php]: Nice :+1:

**Comment by eric-chau** [comment]
Nice @mickaelandrieu :+1:

**Comment by mickaelandrieu** [comment]
thanks for review

**Pull request merged**